src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -1049,29 +1049,55 @@ void ReduceAxesBackwardUseInOut(const nnvm::NodeAttrs& attrs,
   ReduceAxesBackwardUseInOutImpl<xpu, OP, normalize>(ctx, small, inputs, req, outputs);
 }
 
+namespace {  // unnamed namespace to keep scope of the struct within the file
+struct shape_and_stride {
+  index_t in_stride[MXNET_SPECIAL_MAX_NDIM];
+  index_t out_stride[MXNET_SPECIAL_MAX_NDIM];
+  index_t input_shape[MXNET_SPECIAL_MAX_NDIM];
+  index_t output_shape[MXNET_SPECIAL_MAX_NDIM];
+};
+
+inline void PrepareAUXData(shape_and_stride *aux_data,
```

Comment:
How about just rename this function to `CalculateShapeAndStride`?

Reply:
we are only calcualting Stride and storing shape(since its faster to access it this way as compared to mshadow::Shape). Preparing Auxilliary data for execution makes logical sense here

Reply:
I will add description comments on top of this function to explain what it is.